app/controllers/application_controller.rb

```diff
@@ -66,11 +66,13 @@ def current_sp
   end
 
   def sp_from_sp_session
-    ServiceProvider.from_issuer(sp_session[:issuer])
+    sp = ServiceProvider.from_issuer(sp_session[:issuer])
+    sp if sp.is_a? ServiceProvider
```

Comment:
this is avoiding the `NullSessionProvider`?

What if we update `ServiceProvider.from_issuer` to take another argument?

```ruby
ServiceProvider.from_issuer(sp_session[:issuer], null_object: false)
# or maybe
ServiceProvider.from_issuer(sp_session[:issuer], nil_on_not_found: true)
```